tpm/open_other.go

```diff
@@ -0,0 +1,42 @@
+// +build !windows
+
+// Copyright (c) 2019, Google LLC All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tpm
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/google/go-tpm/tpmutil"
+)
+
+// OpenTPM opens a channel to the TPM at the given path. If the file is a
+// device, then it treats it like a normal TPM device, and if the file is a
+// Unix domain socket, then it opens a connection to the socket.
+func OpenTPM(path string) (io.ReadWriteCloser, error) {
+	rwc, err := tpmutil.OpenTPM(path)
+	if err != nil {
+		return nil, err
+	}
+
+	// Make sure this is a TPM 1.2
+	_, err = GetManufacturer(rwc)
```

Comment:
Will this work if the TPM is in startup as well? iirc a specific sequence of commands must be received at startup for initialization to succeed?

Reply:
So this will not work if the TPM has not received the TPM2_Startup command (as that has to be the first command the TPM receives).  However, for a physical TPM this should be sent by the firmware ([example](https://github.com/tianocore/edk2/blob/master/SecurityPkg/Tcg/Tcg2Pei/Tcg2Pei.c#L875-L899)), and for a simulator, it should be sent before go-tpm gets a handle to it ([example](https://github.com/google/go-tpm-tools/blob/5d2fd7f4b3e59d91cba97ab88e8c0777711737a0/simulator/simulator.go#L148-L164)).

Because of this, for instance, the Linux kernel's first command to the TPM is usually `GetCapablity`, not `Startup`. 

Does that seem reasonable?

Reply:
SGTM.